docs/griptape-framework/structures/tasks.md

```diff
@@ -146,3 +146,254 @@ agent.run()
+                             lives become easy. Artificial Intelligence is the  
+                             theory and development of computers, which imitates
+                             the human intelligence and senses, such as visual  
+                             perception, speech recognition, decision-making,   
+                             and translation between languages. Artificial      
+                             Intelligence has brought a revolution in the world 
+                             of technology.                                     
+[10/12/23 17:43:07] INFO     TextSummaryTask ffba7012a1fb4711b8f33a67c08ad677   
+                             Output: Artificial Intelligence (AI) is a branch of
+                             computer science that focuses on creating machines 
+                             capable of thinking and learning. It encompasses   
+                             various fields such as machine learning, neural    
+                             networks, and deep learning. AI has the potential  
+                             to revolutionize sectors like healthcare, finance, 
+                             and transportation. It is important to make        
+                             computers intelligent to make our lives easier. AI 
+                             imitates human intelligence and senses, bringing a 
+                             revolution in technology. 
+```
+
+# Text Query Task
+
+To query a text, you can use the TextQueryTask. This task allows us to pass in a Query Engine and a set of arguments to the engine.
+
+```python
+import os
+from griptape.structures import Agent
+from griptape.tasks import TextQueryTask
+from griptape.drivers import PineconeVectorStoreDriver, OpenAiEmbeddingDriver
+from griptape.engines import VectorQueryEngine
+from griptape.artifacts import TextArtifact
+
+# Initiate Embedding Driver and Vector Store Driver
+embedding_driver = OpenAiEmbeddingDriver()
+vector_store_driver = PineconeVectorStoreDriver(
+            api_key=os.environ["PINECONE_API_KEY"],
+            index_name=os.environ["PINECONE_INDEX_NAME"],
+            environment=os.environ["PINECONE_ENVIRONMENT"],
+            embedding_driver=embedding_driver
+        )
+
+st1 = TextArtifact("Griptape builds AI-powered applications that connect securely to your enterprise data and APIs.")
+st2 = TextArtifact("Griptape Agents provide incredible power and flexibility when working with large language models.")
+
+# Create a VectorQueryEngine using the LocalVectorStoreDriver
+vector_query_engine = VectorQueryEngine(
+    vector_store_driver=vector_store_driver
+)
+
+vector_query_engine.upsert_text_artifact(artifact=st1)
+vector_query_engine.upsert_text_artifact(artifact=st2)
+
+# Instantiate the agent and add TextQueryTask with the VectorQueryEngine
+agent = Agent()
+agent.add_task(
+    TextQueryTask(
+        "{{ args[0] }}",
```

Comment:
Remove `"{{ args[0] }}"` parameter

Reply:
did `Respond to the users following request: {{ args[0] }}`